test-suite/test-squid-conf.sh

```diff
@@ -243,7 +243,9 @@ fi
 
 errorLog="squid-stderr.log"
 
-$sbindir/squid -k parse -f $configFile 2> $errorLog
+configFileDirName=`dirname $configFile`
+configFileName=`basename $configFile`
+(cd $configFileDirName && $sbindir/squid -k parse -f $configFileName) 2> $errorLog
```

Comment:
This fixes a problem where 'squid -k' cannot open some nested configuration files:

    WARNING: file :empty.conf not found

Squid does not do any search when opening configuration files. It uses `ConfigParser::CfgFile::startParse()` to `fopen()` the files (fopen() can locate files relative to the current process directory). The file searching logic exists somewhere else (e.g., readdir() in UFSSwapDir.cc), but these are different areas of code.